libs/gretty/src/main/groovy/org/akhikhl/gretty/ServerConfigWithInputs.java

```diff
@@ -0,0 +1,172 @@
+package org.akhikhl.gretty;
+
+import groovy.lang.Closure;
+import org.gradle.api.model.ReplacedBy;
+import org.gradle.api.tasks.Console;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.InputFile;
+import org.gradle.api.tasks.Internal;
+import org.gradle.api.tasks.Optional;
+import org.gradle.api.tasks.PathSensitive;
+import org.gradle.api.tasks.PathSensitivity;
+
+import java.util.List;
+import java.util.Map;
+
+public interface ServerConfigWithInputs {
+    @Optional
+    @Input
+    List<String> getJvmArgs();
+
+    @Optional
+    @Input
+    Map<String, String> getSystemProperties();
+
+    @Optional
+    @Input
+    String getServletContainer();
+
+    @Optional
+    @Input
+    Boolean getManagedClassReload();
+
+    @Optional
+    @Input
+    String getHost();
+
+    @Optional
+    @Input
+    Boolean getHttpEnabled();
+
+    @Optional
+    @Input
+    Integer getHttpPort();
+
+    @Optional
+    @Input
+    Integer getHttpIdleTimeout();
+
+    @Optional
+    @Input
+    Boolean getHttpsEnabled();
+
+    @Optional
+    @Input
+    Integer getHttpsPort();
+
+    @Optional
+    @Input
+    Integer getHttpsIdleTimeout();
+
+    @Optional
+    @Input
+    String getSslHost();
+
+    @Optional
+    @InputFile
+    @PathSensitive(PathSensitivity.NONE)
+    Object getSslKeyStorePath();
+
+    @Internal
+    String getSslKeyStorePassword();
+
+    @Internal
+    String getSslKeyManagerPassword();
+
+    @Optional
+    @InputFile
+    @PathSensitive(PathSensitivity.NONE)
+    Object getSslTrustStorePath();
+
+    @Internal
+    String getSslTrustStorePassword();
+
+    @Input
+    boolean getSslNeedClientAuth();
+
+    @Internal
+    Object getRealm();
+
+    @Internal
+    Object getRealmConfigFile();
+
+    @Internal
+    Object getServerConfigFile();
+
+    @Optional @Input
+    String getInteractiveMode();
+
+    @Optional @Input
+    Integer getScanInterval();
+
+    @Optional
+    @InputFile
+    @PathSensitive(PathSensitivity.NONE)
+    Object getLogbackConfigFile();
+
+    @Console
```

Comment:
That's cool, I did not know Gradle made that distinction.

Reply:
The annotation is only for documentation.